operator/src/main/java/oracle/kubernetes/operator/helpers/PodHelper.java

```diff
@@ -496,6 +522,13 @@ protected String getClusterName() {
       return new ArrayList<>(super.getContainerCommand());
     }
 
+    @Override
+    protected List<V1Container> getContainers() {
+      List<V1Container> containers = new ArrayList<>(super.getContainers());
+      exporterContext.addContainer(containers);
```

Comment:
It might be more clear to add the sidecar container in `createPodSpec` since the method name `getContainers` is a bit misleading here.

Reply:
The "sidecar" name is really a design pattern, but the pod spec type only has a list of containers. There really isn't any distinction between the main and sidecar containers when you create the pod. So, I think that this name makes sense since it's generating the content for the containers field of the pod spec.

Reply:
ok, yes. Currently, we add the primary container in `createPodSpec` in BasePodStepContext and getContainers() method name implies it's only getting existing containers. I was wondering if the container item for the sidecar container can be added in `createPodSpec` similar to the primary container or change the method name to something like addAndGetContainer().

```
  protected V1PodSpec createPodSpec(TuningParameters tuningParameters) {
    return new V1PodSpec()
        .containers(getContainers())
        .addContainersItem(createPrimaryContainer(tuningParameters))
```

Reply:
No, because it's not under user control. It is an internal implementation detail of the operator.

I agree that `getContainers` is a confusing name, but I wasn't sure what to change it to. I did rename `createContainer` to `createPrimaryContainer` to add clarity. Maybe `getContainers` should be `getUserContainers` or something along those lines.

Reply:
I thought the "sidecar" container for monitoring exporter is not a user container but something that the operator injects. The users might have their own init and/or sidecar containers. In that case, `getUserContainers` name will still be confusing.

Reply:
Correct, and that's why this is confusing. the `getContainers` method returns the list of containers the user has specified in the domain - they have nothing to do with the exporter sidecar container.

Reply:
Should the method be called `getUserContainersAndAddExporterContainer`? I'm ok either way, just thought it might be slightly more clear for someone trying to read the code for the first time.

Reply:
That sounds excessively implementation specific, and problematic. The intent is to add all the containers other than the primary one, and it therefore overrides the default, which is just to add the user-specified containers.

Reply:
Well, in that case, can the method be renamed `getAndAddContainers` or just `addContainers`?